project/data_analysis/python/calibration/get_calibs.py

```diff
@@ -61,66 +62,121 @@
-    array = f"dr6_{ar}"
-    results_dict[array] = {}
-    ref_array = ref_arrays[array]
-
-    spectra_order = [(array, array, "TT"),
-                     (array, ref_array, "TT"),
-                     (ref_array, ref_array, "TT")]
-
-    # Load spectra and cov
-    ps_dict = {}
-    cov_dict = {}
-    for i, (ar1, ar2, m1) in enumerate(spectra_order):
-        _, ps = so_spectra.read_ps(f"{spec_dir}/Dl_{ar1}x{ar2}_cross.dat",
-                                   spectra = spectra)
-        ps_dict[ar1, ar2, m1] = ps[m1]
-        for j, (ar3, ar4, m2) in enumerate(spectra_order):
-            if j < i: continue
-            cov = np.load(f"{cov_dir}/analytic_cov_{ar1}x{ar2}_{ar3}x{ar4}.npy")
-            cov = so_cov.selectblock(cov, modes, n_bins = n_bins, block = m1 + m2)
-            cov_dict[(ar1, ar2, m1), (ar3, ar4, m2)] = cov
-
-    # Concatenation
-    spec_vec, full_cov = consistency.append_spectra_and_cov(ps_dict, cov_dict, spectra_order)
-
-    # Save and plot residuals before calibration
-    res_spectrum, res_cov = consistency.project_spectra_vec_and_cov(spec_vec, full_cov, proj_pattern)
-    np.savetxt(f"{residual_output_dir}/residual_{array}_before.dat", np.array([lb, res_spectrum]).T)
-    np.savetxt(f"{residual_output_dir}/residual_cov_{array}.dat", res_cov)
-
-    lmin, lmax = multipole_range[array]
-    id = np.where((lb >= lmin) & (lb <= lmax))[0]
-    consistency.plot_residual(lb, res_spectrum, {"analytical": res_cov}, "TT", array,
-                              f"{residual_output_dir}/residual_{array}_before",
-                              lrange=id, l_pow=1)
-
-    # Calibrate the spectra
-    cal_mean, cal_std = consistency.get_calibration_amplitudes(spec_vec, full_cov,
-                                                    proj_pattern, "TT", id,
-                                                    f"{chains_dir}/{array}")
-
-    results_dict[array] = {"multipole_range": multipole_range[array],
-                           "ref_array": ref_arrays[array],
-                           "calibs": [cal_mean, cal_std]}
-
-    calib_vec = np.array([cal_mean**2, cal_mean, 1])
-    res_spectrum, res_cov = consistency.project_spectra_vec_and_cov(spec_vec, full_cov,
-                                                          proj_pattern,
-                                                          calib_vec = calib_vec)
-    np.savetxt(f"{residual_output_dir}/residual_{array}_after.dat", np.array([lb, res_spectrum]).T)
-    consistency.plot_residual(lb, res_spectrum, {"analytical": res_cov}, "TT", array,
-                              f"{residual_output_dir}/residual_{array}_after",
-                              lrange=id, l_pow=1)
+
+
+tests = ["AxA-AxP", "AxA-PxP", "PxP-AxP"]
+
+for test in tests:
+    for ar in d["arrays_dr6"]:
+    
+        map_set = f"dr6_{ar}"
+        ref_map_set = ref_map_sets[map_set]
+
+        name, proj_pattern = get_proj_pattern(test, map_set, ref_map_set)
+
+        results_dict[name] = {}
```

Comment:
this line is not needed since it is overloaded by https://github.com/simonsobs/PSpipe/pull/159/files#diff-1e4eec3c4d38601d8231c75e34f5ebdc647face8396a65d940dc0c9bc3a3c816R135